tests/Unit/NumericalAlgorithms/LinearOperators/Test_CoefficientTransforms.cpp

```diff
@@ -0,0 +1,125 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/ModalVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"
+#include "NumericalAlgorithms/LinearOperators/CoefficientTransforms.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/Gsl.hpp"
+
+// These tests generate a function `u_expected` from a linear superposition of
+// the basis functions, which are then transformed to spectral space
+// (u_coeff) and the coefficients compared with the coefficients set in physical
+// space. The functions are then transformed back (u_back) and compared to the
+// original function, `u_expected`.
+namespace {
+template <Spectral::Basis Basis, Spectral::Quadrature Quadrature, size_t Dim>
+void check_transforms(
+    const Mesh<Dim>& mesh,
+    const std::vector<std::array<size_t, Dim>>& coeffs_to_include) noexcept {
+  CAPTURE(Basis);
+  CAPTURE(Quadrature);
+
+  const auto logical_coords = logical_coordinates(mesh);
+  DataVector u_expected(mesh.number_of_grid_points(), 0.0);
+  for (const auto& coeffs : coeffs_to_include) {
```

Comment:
I think it may be clearer to make the loop variable singular (`coeffs` -> `coeff`). While it's true the coeff has multiple components, it does represent a single coefficient in the spectral expansion. I take the blame for making this plural in my code snippet :)